scripts/generate-index-json.ts

```diff
@@ -0,0 +1,174 @@
+    return value !== undefined;
+}
+
+function initialiseGitHubApi() {
+    let authToken: string | undefined = undefined;
+    const args = process.argv.slice(2);
+    if (args[0] === '--authToken' && args[1]) {
+        authToken = args[1];
+    } else {
+        authToken = process.env.AUTH_TOKEN;
+    }
+
+    if (!authToken) {
+        console.warn(
+            colours.yellow(
+                'No auth token was provided, so you may encounter rate limit issues when calling the GitHub API.',
+            ),
+        );
+        console.warn(
+            colours.yellow(
+                'You can provide an auth token either through the "--authToken" parameter or by setting the "AUTH_TOKEN" environment variable.\n',
+            ),
+        );
+    }
+
+    return new Octokit({ request: { fetch }, auth: authToken });
+}
+
+const octokit = initialiseGitHubApi();
+
+async function readJsonFiles(): Promise<ParsedOrgFile[]> {
+    const indexDir = path.join(__dirname, '..', 'index');
+    const indexFiles = await fs.readdir(indexDir);
+
+    return Promise.all(
+        indexFiles
+            .filter((f) => f.endsWith('.json'))
+            .map(async (f) => ({
+                id: f.replace('.json', ''),
+                orgIndex: JSON.parse(await fs.readFile(path.join(indexDir, f), 'utf-8')),
+            })),
+    );
+}
+
+async function generateIndex(orgIndices: ParsedOrgFile[]): Promise<AppIndex> {
+    const appIndex: AppIndex = { orgs: {}, apps: [] };
+
+    const data = await Promise.all(orgIndices.map(fetchOrgData));
+    for (const { org, apps } of data.filter(notUndefined)) {
+        appIndex.orgs[org.id] = org;
+        appIndex.apps.push(...apps);
+    }
+
+    return appIndex;
+}
+
+async function fetchOrgData({
+    id: orgId,
+    orgIndex,
+}: ParsedOrgFile): Promise<{ org: Organization; apps: Application[] } | undefined> {
+    try {
+        const userData = await octokit.users.getByUsername({ username: 'id' });
+
+        console.log(colours.green(`Fetched data for user ${orgId}`));
+
+        const org: Organization = {
+            id: orgId,
+            name: orgIndex.name,
+            description: orgIndex.description,
+            isPartner: partnerIds.includes(orgId),
+            type: userData.data.type as (typeof validOrgKinds)[number],
+            urls: {
+                support: userData.data.url,
+                email: userData.data.email ?? undefined,
+                blog: userData.data.blog ?? undefined,
+                twitter: userData.data.twitter_username ?? undefined,
+            },
+            avatar: userData.data.avatar_url,
+            location: userData.data.location ?? undefined,
+        };
+
+        const apps = await Promise.all(orgIndex.apps.map((app) => fetchRepoData(orgId, app)));
+
+        return { org, apps: apps.filter(notUndefined) };
+    } catch (e) {
+        console.warn(colours.yellow(`Failed to fetch account data for user "${orgId}"`));
```

Comment:
We'd want to block contributions that add invalid entries, which would include orgs that don't exist. I think this exception should crash the script to avoid this going into main